components/patient-data/ui/src/main/resources/PhenoTips/PatientConsent.xml

```diff
@@ -384,33 +406,22 @@
           }).get();
         },
 
-        attachManagingListeners: function (baseUrl)
+        attachStateChangeCallbacks: function(callback)
         {
           var _this = this;
           var consentList = $(".consents");
           consentList.on("change", "li input", function (event)
           {
             var checkbox = $(event.target);
-            var rest = _this.interactWithManagingRest(checkbox.attr("data-bind"), baseUrl);
+            var name = checkbox.attr("data-bind");
             if (checkbox.is(":checked")) {
-              rest("grant");
+              callback(name, true);
             } else {
-              rest("revoke");
+              callback(name, false);
```

Comment:
I wanted to separate UI implementation form the logic, e.g. make it clear for un-initiated (like myself) that checkbox.attr("data-bind") is actually the name of the consent. Would be easier if we had named functional parameters in JS, e.g. callback( name:checkbox.attr("data-bind"), status: checkbox.is(":checked:") )